src/frontend/js/modules/application-menu.js

```diff
@@ -1,131 +1,40 @@
-      var isActive = defaultValue;
-
-      if (angular.isDefined(flag)) {
-        isActive = featureFlags.isOn(flag) === undefined ? defaultValue : featureFlags.isOn(flag);
-      }
-
-      if (!isActive) {
-        return '';
-      }
-
-      if (icon.url) {
-        iconURL = icon.url;
-      } else if (icon.name) {
-        iconURL = _.template(iconUrlTemplate)({icon: icon.name});
-      }
-
-      if (angular.isObject(href)) {
-        aHref = href.url;
-        aTarget = href.target || '';
-        aRel = href.rel || aRel;
-      } else {
-        aHref = href;
-        aTarget = '';
-      }
-
-      return iconURL && iconURL.match(/.\S+\.svg/) ?
-      _.template(svgTemplate)({ href: aHref, target: aTarget, rel: aRel, iconURL: iconURL, label: label }) :
-      _.template(template)({ href: aHref, target: aTarget, rel: aRel, iconURL: iconURL, label: label });
-    };
-  })
-
-  .directive('applicationMenuToggler', function($rootScope, $document, $popover,
-    POPOVER_APPLICATION_MENU_OPTIONS, APP_MENU_OPEN_EVENT) {
+// TODO: Write tests for this
+angular.module('esn.application-menu', ['op.dynamicDirective'])
+  .directive('applicationMenuToggler', function() {
     return {
       restrict: 'E',
       scope: true,
       replace: true,
       template: require("../../views/modules/application-menu/application-menu-toggler.pug"),
-      link: function(scope, element) {
-        var backdrop = angular.element('<div id="application-menu-backdrop" class="modal-backdrop in visible-xs">'),
-        body = $document.find('body').eq(0),
-            popover = $popover(element, POPOVER_APPLICATION_MENU_OPTIONS);
-
-            popover.$scope.$on('application-menu.show.before', function() {
-              body.append(backdrop);
-              $rootScope.$broadcast(APP_MENU_OPEN_EVENT);
-              scope.$applyAsync();
-            });
-
-            popover.$scope.$on('application-menu.hide.before', function() {
-              backdrop.remove();
-              scope.$applyAsync();
-            });
-
-            element.click(popover.toggle.bind(null, null));
-      }
-    };
-  })
-
-  .directive('forceCloseOnLinksClick', function($timeout) {
-    return {
-      restrict: 'A',
-      link: function(scope, element) {
-        $timeout(function() {
-          element.find('a').click(scope.$parent.$hide.bind(null, null));
-        }, 0, false);
-      }
-    };
-  })
-
-  .directive('forceMarginLeft', function($timeout) {
-    return {
-      restrict: 'A',
-      link: function(scope, element, attrs) {
-        $timeout(function() {
-          var offset = element.offset();
-
-          element.offset({top: offset.top, left: offset.left - attrs.forceMarginLeft});
-        }, 0, false);
+      link: function($scope) {
+        const defaultAppGridItems = JSON.stringify([
+          {
+            name: 'Calendar',
+            url: 'http://localhost:9900/#/calendar'
```

Comment:
Should we have `.env.default` to store these default values. Then `.env`, `.env.dev` `.env.prod` to override it.

Reply:
.env.default is a rare convention. Usually, we have env.example then in each environment, the developer just copies the values to corresponding .env

Reply:
In the other PR, you can see that I already provide the default values via `.env.example`, but it still requires the dev to remember to create `.env` with those values. If the dev forgets to do so, the application grid will be blank, and this part of code is for that case. Maybe we can neglect that problem and remove this part of the code, WDYT?

Reply:
Alright I have removed the default values in JS. Each developer will now have to remember to create the `.env` file or the application grid will be an empty shell.

Reply:
I'm not against the default one. Just the way to store these default values.

At least we need to document it somewhere, log an alert if the app grid is not defined yet.

Reply:
The documentation part will be in each SPA's README just like how we've been documenting the usage of `OPENPAAS_URL`. Logging a warning/error is a nice idea also.

Reply:
Cool!